### DISCUSSION TITLE
How to stop Datagrid editing event to propagate?

### DISCUSSION THREAD

**Discussion by abraaoprogramador** [AvaloniaUI/Avalonia#12769] [Q&A]
Hi! I'm using Avalonia and a Realm database. I have a Datagrid and I want to display and edit a list of IRealmObject. This type of object reacts to changes automatically. But if I need to make changes, the changes need to be inside of a realm.Write(). I want the Datagrid to react to the edition of the Datagrid to use that method. The problem is that I don't know how to.

Firstly, I tried the this.RaiseAndSetIfChanged(ref _Property, value) with a Subscribe but it doesn't work as I needed. After that, I tried to use the get and set of the property but it doesn't work either. Then, I used CellEditEnding on the code-behind but the Avalonia doesn't use an event.Handled so after the editing, the event still occurs so the Realm throws an exception.

Lastly, I tried to use

`<KeyBinding Gesture="Enter" Command="{Binding EnterCommand}" />` 

But after I treated the command, Realm still throws an exception.

What I understood after this experience was that I can't block the event to pass to another step so Avalonia is still raising the event after my methods.

My question is: how do I treat the editing event and block the propagation?

The Realm exception is Realms.Exceptions.RealmInvalidTransactionException.

**Comment by timunie** [comment]
iirc there is an editendingevent in the datagrid which you can probably listen to. I have no idea about the Realm database. 

Another idea could be to have a seperate ViewModel for editing which triggers the updates to the DB as needed.

**Reply by abraaoprogramador** [reply]
The problem with EditEndingEvent is that after the execution of my code, the event propagates so it causes an error.

**Reply by timunie** [reply]
Is there no `e.Handled` switch?

**Reply by abraaoprogramador** [reply]
Yes, there is no e.Handled switch, the compiler does not find an e.Handled on the CellEditEndingEvent. Generally, I use e.Handled to stop events from propagating.

Edited: I clarified my comment.

**Reply by abraaoprogramador** [reply]
Is the missing e.Handled a bug?

**Reply by timunie** [reply]
probably a feature request. Bug I am afraid that this will be for 12.0 only valid as it may be a breaking change (new type of Event argeuments). But worth to open a feature request. Can you do it?
